pylace/src/utils.rs

```diff
@@ -559,44 +559,46 @@ fn df_to_values(
     indexer: &Indexer,
     engine: &lace::Engine,
 ) -> PyResult<DataFrameComponents> {
-    let row_names = if df.hasattr("index")? {
-        let index = df.getattr("index")?;
-        srs_to_strings(index).ok()
-    } else {
-        df.call_method1("__getitem__", ("index",))
-            .and_then(srs_to_strings)
-            .ok()
-    };
-
     Python::with_gil(|py| {
-        let (columns, data) = {
+        let (columns, data, row_names) = {
             let columns = df.getattr("columns").unwrap();
             if columns.get_type().name().unwrap().contains("Index") {
                 // Is a Pandas dataframe
+                let index = df.getattr("index")?;
```

Comment:
Just curious: is the difference between using "index" on this line and "Index" on the previous one significant?

Reply:
Let me know if there is a better way to do this put the answer is that `pandas.DataFrame.index` (lowercase 
"i") gets the index of the dataframe which has type `Index` (capital "I"). polars purposefully doesn't have an index, so this is a simple way to distinguish what type of DataFrame we're working with.